repeatmasking_utils.py

```diff
@@ -0,0 +1,663 @@
+# See the NOTICE file distributed with this work for additional information #pylint: disable=missing-module-docstring
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import multiprocessing
+import os
+import pathlib
+import re
+import subprocess
+
+from pathlib import Path
+
+from typing import Union
+from utils import (
+    create_dir,
+    check_exe,
+    check_gtf_content,
+    get_seq_region_lengths,
+    create_slice_ids,
+    slice_output_to_gtf,
+    get_sequence,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def run_repeatmasker_regions(  # pylint: disable=too-many-arguments
+    genome_file: Union[pathlib.Path, str],
+    repeatmasker_path: str,
+    library: str,
+    species: str,
+    main_output_dir: str,
+    num_threads: int,
+):
+    """
+    Run Repeatmasker on genomic slices using the crossmatch engine.
+    Args:
+        genome_file : pathlib.Path
+        repeatmasker_path : str path to the RepeatMasker executable
+        library : str
+        species :str
+        main_output_dir : pathlib.Path
+        num_threads: int
+
+    Return:
+        A GTF file with the repeatmasked sequence for each genome slice
+
+    """
+    if not repeatmasker_path:
+        repeatmasker_path = "RepeatMasker"
+
+    # if not library:
+    #    library = "homo"
+
+    check_exe(repeatmasker_path)
+    repeatmasker_output_dir = Path(create_dir(main_output_dir, "repeatmasker_output"))
+    os.chdir(repeatmasker_output_dir)
+
+    output_file = repeatmasker_output_dir / "annotation.gtf"
+    if output_file.exists():
+        transcript_count = check_gtf_content(output_file, "repeat")
+        if transcript_count > 0:
+            logger.info("Repeatmasker gtf file exists")
+            return
+
+    logger.info("Creating list of genomic slices")
+    seq_region_lengths = get_seq_region_lengths(genome_file, 5000)
+    slice_ids = create_slice_ids(
+        seq_region_lengths, slice_size=1000000, overlap=0, min_length=5000
+    )
+
+    if not library:
+        if not species:
+            species = "homo"
+            generic_repeatmasker_cmd = [
+                repeatmasker_path,
+                "-nolow",
+                "-species",
+                species,
+                "-engine",
+                "crossmatch",
```

Comment:
We should be consistent with the update that we now use the RM blast engine https://github.com/Ensembl/ensembl-analysis/pull/365/files
But As Thibaut mentioned in another comment, we should provide the engine as a an argument and have RM blast as the default

Reply:
It is a default parameter in the config json where all the paths for the softwares

Reply:
set RMblast as default in the config json where all the path for software and some params have been moved The paths for species and library are still arguments for main because it is easier set up them there wen we run the command from the guiHive pipeline